sdk/python/kfp/dsl/for_loop.py

```diff
@@ -188,6 +191,13 @@ def from_pipeline_channel(
         compilation progress in cases of unknown or missing type
         information.
         """
+        if not isinstance(channel, for_loop.LoopArgumentVariable):
```

Comment:
For my own understanding, is this condition needed for nested loops? Perhaps add a comment to indicate its purpose.

Can we ensure we have test coverage for both cases (i.e., `True` and `False`)?

Reply:
Per offline convo, ignoring nested loop case (channel is LoopArgumentVariable). Added todo